girder/girder_large_image/rest/large_image_resource.py

```diff
@@ -449,56 +449,100 @@ def _deleteCachedImages(self, spec, associatedImages=False, imageKey=None):
+        .param('cancelJobs', 'Whether unfinished large image job(s) associated '
+               'with items in the folder should be canceled, then a new large '
+               'image created. If false, items with an unfinished large image '
+               'will be skipped.', required=False, default=False,
+               dataType='boolean')
+        .param('redoExisting', 'Whether existing large images should be removed and '
+               'recreated.', required=False, default=False, dataType='boolean')
         .errorResponse('ID was invalid.')
         .errorResponse('Write access was denied for the folder.', 403),
     )
     def createLargeImages(self, folder, params):
         user = self.getCurrentUser()
-        createJobs = 'always' if self.boolParam('force', params, default=False) else True
-        return self.createImagesRecurseOption(folder=folder, createJobs=createJobs, user=user,
-                                              recurse=params.get('recurse'),
-                                              localJobs=params.get('localJobs'))
-
-    def createImagesRecurseOption(self, folder, createJobs, user, recurse, localJobs):
+        createJobs = params.get('createJobs')
+        if createJobs in ['true', 'false']:
+            createJobs = self.boolParam('createJobs', params, default=False)
+        return self.createLargeImagesRecurse(folder=folder, user=user,
+                                             recurse=params.get('recurse'), createJobs=createJobs,
+                                             localJobs=params.get('localJobs'),
+                                             cancelJobs=params.get('cancelJobs'),
+                                             redo=params.get('redoExisting'))
+
+    def createLargeImagesRecurse(
+            self, folder, user, recurse, createJobs, localJobs, cancelJobs, redo):
         result = {'childFoldersRecursed': 0,
                   'itemsSkipped': 0,
+                  'jobsCanceled': 0,
+                  'jobsFailedToCancel': 0,
                   'largeImagesCreated': 0,
                   'largeImagesRemovedAndRecreated': 0,
                   'totalItems': 0}
         if recurse:
             for childFolder in Folder().childFolders(parent=folder, parentType='folder'):
                 result['childFoldersRecursed'] += 1
-                childResult = self.createImagesRecurseOption(folder=childFolder,
-                                                             createJobs=createJobs, user=user,
-                                                             recurse=recurse, localJobs=localJobs)
+                childResult = self.createLargeImagesRecurse(folder=childFolder, user=user,
+                                                            recurse=recurse, createJobs=createJobs,
+                                                            localJobs=localJobs,
+                                                            cancelJobs=cancelJobs, redo=redo)
```

Comment:
This feels logically backwards to me. I think I would expect:
```
def createLargeImage(self, folder):
    for item in folder:
        ***createImage logic here***
 
def folderRecurse(self, params):
     if recurse:
        for childFolder in folder:
           createLargeImage()
```
additionally I am not sure  I would make a new function to check the recurse param. That is a design choice for you and David to decide though.